packages/manager/apps/web/client/app/email-domain/order/order.routing.js

```diff
@@ -0,0 +1,78 @@
+import filter from 'lodash/filter';
+import includes from 'lodash/includes';
+import map from 'lodash/map';
+import union from 'lodash/union';
+import { PRODUCT_NAME } from './order.constants';
+import EmailDomainOffer from './EmailDomainOffer.class';
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('app.mx-plan', {
+    url: '/configuration/mx_plan?domain',
+    component: 'emailDomainOrder',
+    params: {
+      domain: null,
+    },
+    onExit: /* @ngInject */(cart, OrderService) => OrderService.deleteCart(cart),
+    resolve: {
+      addConfiguration: /* @ngInject */ (
```

Comment:
There are lots of function defined in the routing like 'addConfiguration', 'deleteItem' ...
Imho i don't think they belong here, it should exist in the controller or if you need to reuse these, in a service.
Our routing is starting to look like a generic placeholder and the resolve part should only be used for routing related stuff or inputs passed to the component.

Reply:
I don't agree with you. There are some examples when putting this stuff in the controller will lead to duplication. Let's says you have two data provider but want the same display. If you don't pass it as a binding (in the resolve's routing in our case) you'll need to duplicate the two components because you'll have to handle a controller by data provider

Reply:
Then you can use a service if you want but there is no way that the routing file is the proper way to avoid code duplication.

Also i think it's wrong to wrap a simple function call in another function like that :
```
getCheckout: /* @ngInject */ (
        cart,
        OrderService,
      ) => () => OrderService.getCheckoutInformations(cart)
```
you are not factoring anything it's useless, the function you are calling is already coming from a service, if changes have to be made then the service will be updated

Reply:
A good example is the PCI containers https://github.com/ovh/manager/tree/master/packages/manager/modules/pci/src/projects/project/storages/containers
You have two different behaviours/API providers but the component is barely aware of it. It just worries about displaying stuff. 

I'm having trouble understanding how can an injected service could prevent duplication if you need different data providers

Reply:
IMO, it's a good implementation. 
If we read the code, `cart` is never used in controller or template, because the main goal of this component is to provide a way to order a product whatever what should be used/done to achieve it. 
But yes, some `resolve` methods decorate service's methods but they also provide an easy way to order the product because It's not the component's responsibility to manage the cart.

Abstraction helps us to have reusable components.

Reply:
i see what you mean, it's still possible to pass a service to the component, something like :
```
resolve : {
  containerApi: (MySpecificContainerApi) => MySpecificContainerApi,
}
```
or yes you can bind every function you need in the routing, maybe it's a matter of preferences

Reply:
you just made me think that something like this is possible too :
```
resolve: {
  cartApi: (OrderService) => {
    addConfiguration: (cartId) => OrderService.addConfiguration(cartId),
    addItem: (itemId) => OrderService...,
    [...]
  },
}
```
that way we are creating a bridge to OrderService and it makes the component interface cleaner i think

i'm just throwing some ideas, no need to make changes